src/maliput_malidrive/builder/builder_tools.cc

```diff
@@ -292,20 +292,77 @@ LaneTravelDirection::LaneTravelDirection(const std::optional<std::string>& user_
   }
   const char* travel_dir_attribute = vector_lane_element->Attribute(kTravelDirTag);
   if (travel_dir_attribute == nullptr) {
-    travel_dir_ = LaneTravelDirection::Direction::kUndefined;
-    return;
+    return travel_direction;
   }
-  travel_dir_ = str_to_direction(travel_dir_attribute);
+  travel_direction.travel_dir_ = travel_direction.str_to_direction(travel_dir_attribute);
+  return travel_direction;
+}
+
+LaneTravelDirection LaneTravelDirection::CreateFromLaneGroupDirection(
+    int lane_id, const std::optional<xodr::Lane::Direction>& hand_traffic_rule_direction,
+    const std::optional<xodr::RoadHeader::HandTrafficRule>& hand_traffic_rule) {
+  if (!hand_traffic_rule_direction.has_value()) {
+    return LaneTravelDirection(LaneTravelDirection::Direction::kUndefined);
+  }
+  auto lane_direction = hand_traffic_rule_direction.value();
+  if (lane_direction == xodr::Lane::Direction::kBoth) {
+    return LaneTravelDirection(LaneTravelDirection::Direction::kBidirectional);
+  }
+  LaneTravelDirection travel_direction = CreateFromHandTrafficRule(lane_id, hand_traffic_rule);
+  if (lane_direction == xodr::Lane::Direction::kStandard) {
+    return travel_direction;
+  }
+  if (lane_direction == xodr::Lane::Direction::kReversed) {
+    if (travel_direction.travel_dir_ == LaneTravelDirection::Direction::kForward) {
+      travel_direction.travel_dir_ = LaneTravelDirection::Direction::kBackward;
+    } else if (travel_direction.travel_dir_ == LaneTravelDirection::Direction::kBackward) {
+      travel_direction.travel_dir_ = LaneTravelDirection::Direction::kForward;
+    }
+  }
+  return travel_direction;
+}
+
+LaneTravelDirection LaneTravelDirection::CreateFromHandTrafficRule(
+    int lane_id, const std::optional<xodr::RoadHeader::HandTrafficRule>& hand_traffic_rule) {
+  LaneTravelDirection travel_direction(LaneTravelDirection::Direction::kUndefined);
+  if (!hand_traffic_rule.has_value()) {
+    // No rule is treated as RHT.
+    if (lane_id < 0) {
+      travel_direction = LaneTravelDirection(LaneTravelDirection::Direction::kForward);
+    } else {
+      travel_direction = LaneTravelDirection(LaneTravelDirection::Direction::kBackward);
+    }
+    return travel_direction;
+  }
+  switch (hand_traffic_rule.value()) {
+    case xodr::RoadHeader::HandTrafficRule::kLHT: {
+      if (lane_id > 0) {
+        travel_direction = LaneTravelDirection(LaneTravelDirection::Direction::kForward);
+      } else {
+        travel_direction = LaneTravelDirection(LaneTravelDirection::Direction::kBackward);
+      }
+      break;
+    }
+    case xodr::RoadHeader::HandTrafficRule::kRHT:
+      [[fallthrough]];
+    // Default is treated as RHT.
+    default: {
+      if (lane_id < 0) {
+        travel_direction = LaneTravelDirection(LaneTravelDirection::Direction::kForward);
+      } else {
+        travel_direction = LaneTravelDirection(LaneTravelDirection::Direction::kBackward);
+      }
+    }
+  }
+  return travel_direction;
```

Comment:
I think we can reduce this 30 lines of code to less than 10.
See https://en.cppreference.com/w/cpp/utility/optional/value_or

Reply:
```suggestion
  const auto rule = hand_traffic_rule.value_or(xodr::RoadHeader::HandTrafficRule::kRHT);
  const bool is_forward = (rule == xodr::RoadHeader::HandTrafficRule::kLHT)
                              ? (lane_id > 0)
                              : (lane_id < 0);
  return LaneTravelDirection(is_forward ? Direction::kForward : Direction::kBackward);
```

Reply:
Changed, thanks!